File: LearningProject/Models/InternalUser.cs
using System.ComponentModel.DataAnnotations;
namespace LearningProject.Models
{
public class InternalUser
{
    public int Id { get; set; } 

    [Required]
    public required string Username { get; set; }

    public string? PasswordHash { get; set; }

    public int? DepartmentId { get; set; }
    public Department? Department { get; set; }    

    public int UserId { get; set; }
    public User? User { get; set; }
    public ICollection<InternalUserRole>? InternalUserRoles { get; set; }
}
}

File: client/src/app/core/models/InternalUser.ts
import { Department } from "./Department";
import { InternalUserRole } from "./InternalUserRole";
import { User } from "./User";

export interface InternalUser{
    id : number;
    username:string;
    passwordHash? : string;
    departmentId? : number;
    department? : Department;
    userId : number;
    user: User;
    internalUserRoles? : InternalUserRole[] | null;
}